### PR TITLE
feature/sup-65-car-page

### PR DESCRIPTION
**Pull request by alesiahlushakova**

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -85,10 +87,17 @@ public int changeActiveStatus(Long id, Boolean status) {
...
+        return userRepository.setStatus(true, id);
+    }
+
+
     @Override
     @Transactional
     public int changePassword(Long id, String password) {
-        return userRepository.changePassword(password, id);
+        return userRepository.changePassword(passwordEncoder.encode(password), id);
     }
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/UserServiceImpl.java:107]: You encode the password as "password", you should get the value without quotes and encode it

File: src/main/java/by/itech/lab/supplier/domain/Address.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
     private String addressLine1;
     @Column(nullable = false, name = "address_line_2")
     private String addressLine2;
-    @ManyToOne(cascade = CascadeType.MERGE)
+    @ManyToOne(cascade = CascadeType.ALL)
     @JoinColumn(name = "state_id")
     @ToString.Exclude
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/Address.java]: you shouldn't create state

File: src/main/java/by/itech/lab/supplier/controller/UserController.java [hunk 3]
```diff
@@ -75,14 +75,14 @@ public UserDto getUseByUsername(@PathVariable String username) {
         log.debug("request to get User : {}", username);
         return userService.findByUsername(username);
     }
-    @AdminAccess
+
     @ResponseStatus(HttpStatus.OK)
     @PutMapping(ApiConstants.URL_ID_PARAMETER + ApiConstants.URL_PASSWORD_PARAMETER)
     public int changePassword(@PathVariable Long id, @RequestBody String password) {
         return userService.changePassword(id, password);
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/controller/UserController.java]: You should be protecting this method, but not @AdminAccess. Now any user can change the password to any user of the company which owns. But the user must have opportunity to change only his password.

File: frontend/src/Header.js [hunk 4]
```diff
@@ -52,13 +53,14 @@ function Header() {
...
             </Navbar.Collapse>
         </Navbar>
-    );
-
+    )
 }
 
 
-export default Header
+export default Header
```
Comment (IrynaZazyba) [inline frontend/src/Header.js]: empty line

File: frontend/src/pages/car/ModalAddCar.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    const isValid = (param) => errors.validationErrors.includes(param) ? "form-control is-invalid" : "form-control";
+
+    const addCarHandler = (e) => {
+        e.preventDefault();
+        setErrors(preState => ({
+            ...preState,
+            validationErrors: ''
+        }));
+        console.log(carDto);
+        fetch(`/customers/${currentCustomerId}/car`, {
```
Comment (IrynaZazyba) [inline frontend/src/pages/car/ModalAddCar.js]: Need to check empty fields

**Review by IrynaZazyba — CHANGES_REQUESTED** [review]
Please, correct the layout.

File: src/main/java/by/itech/lab/supplier/controller/UserController.java [hunk 6]
```diff
@@ -97,6 +98,12 @@ public void changeActive(@PathVariable Long id, @RequestBody boolean status) {
         userService.changeActiveStatus(id, status);
     }
 
+    @ResponseStatus(HttpStatus.OK)
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/UserController.java]: Which status is used by default? and anyone will be able to change the status to anyone?

File: src/main/java/by/itech/lab/supplier/controller/UserController.java [hunk 7]
```diff
@@ -71,18 +72,18 @@ public UserDto createUser(@Valid @RequestBody UserDto userDto) {
 
 
     @GetMapping(ApiConstants.URL_USERNAME_PARAMETER)
-    public UserDto getUseByUsername(@PathVariable String username) {
+    public UserDto getUserByUsername(@PathVariable String username) {
         log.debug("request to get User : {}", username);
         return userService.findByUsername(username);
     }
-    @AdminAccess
+
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/UserController.java:78]: so, anyone will be able to change the password to anyone?

**Review by svareyko — CHANGES_REQUESTED** [review]

File: src/main/java/by/itech/lab/supplier/controller/UserController.java (hunk 7 above)
Comment (alesiahlushakova) [reply src/main/java/by/itech/lab/supplier/controller/UserController.java:78]: Each user is able to change his password

Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/controller/UserController.java:78]: here I see the code, which says that I as UserA able to change the password for UserB (or even admin) if I only know his ID

Comment (alesiahlushakova) [reply src/main/java/by/itech/lab/supplier/controller/UserController.java:78]: i've added in service method check if user principal id is the same as in url

File: frontend/src/App.js [hunk 8]
```diff
@@ -63,6 +63,12 @@ function App() {
         })}/>
     };
 
+    const renderCar = () => {
+       return <ProtectedComponent conditions={user.role === "ROLE_SYSTEM_ADMIN"} render={(() => {
+            return <Cars/>
```
Comment (IrynaZazyba) [inline frontend/src/App.js:68]: ADMIN manages the customer's cars, add his role as allowed

File: src/main/java/by/itech/lab/supplier/controller/UserController.java (hunk 7 above)
Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/controller/UserController.java:78]: its not necessary actions, you could use something like:
@PreAuthorize("principal.id == :id")
// not sure in syntax, but you can check how to do it correctly

**Review by IrynaZazyba — APPROVED** [review]

**Review by alexeyoverchenko — APPROVED** [review]

**Review by svareyko — APPROVED** [review]

**Pull request merged**